json-2/src/main/java/com/baeldung/jsonld/deserialization/jsonldjava/jackson/JacksonDeserializationExample.java

```diff
@@ -0,0 +1,40 @@
+package com.baeldung.jsonld.deserialization.jsonldjava.jackson;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.jsonldjava.core.JsonLdOptions;
+import com.github.jsonldjava.core.JsonLdProcessor;
+import com.github.jsonldjava.utils.JsonUtils;
+
+public class JacksonDeserializationExample {
+    public static void main(String[] args) throws IOException {
```

Comment:
I'm not a big fan of these *Example type classes with the main method. Let's move this into a unit test using the test method names (givenX_whenY_thenZ) as mentioned in the author guidelines.

In general, we want as much coverage as possible on our new code. Putting it in a unit test is also easier for readers to run the code and also we validate that it is working as expected.

Reply:
@JonCook I've converted everything into unit tests, but I'm not confident that I've found good names for these tests. Using assertions is definitely better than console prints, so I am happy in that regard, however, the Eclipse formatter doesn't allow multiline strings, so the inline JSON objects look terrible.

Reply:
I think the test approach is good! Could we shorten the test names a little? I think you are on the right track. e.g
`givenAJsonLdObject_whenCompactIsUsedWithEmptyContext_thenItCanBeDeserializedIntoAJacksonAnnotatedJavaObject()`

Could become for example (just an idea):
`givenAJsonLdString_whenCompactIsUsedWithEmptyContext_thenDeserializedCorrectly()`

Let's also please move the Person class into the main src tree.

Finally, I agree the inline JSON does not look good. I'm afraid we will have to fix that manually without using the formatter.

Reply:
@JonCook is `givenAJsonLdObject_whenCompactIsUsedWithEmptyContext_thenItCanBeDeserializedWithJackson()` okay? Are the two other test names okay like they are?